File: rookieone/ruby/spec/day1/part2_spec.rb
require "./code/day1/part2"
require "./code/day1/file_reader"

RSpec.describe "Day 1 Part 2" do
  it "example should be 7" do
    input = Day1FileReader.read_file("./spec/day1/example.txt")

    answer = Day1Part2.new(input).solve()

    expect(answer).to   eq(5)
  end

  it "input should solve the problem" do
    input = Day1FileReader.read_file("./spec/day1/input.txt")

    answer = Day1Part2.new(input).solve()

    expect(answer).to   eq(1822)
  end
end

File: rookieone/node/src/day1/part2_spec.js
import assert from 'assert';

import { measureDepthWindowIncreases } from './part2';
import { readPuzzleInput } from './readPuzzleInput';

describe("Day 1", () => {
  describe("Part 2", () => {
    it("example should return answer", () => {
      const input = readPuzzleInput('./src/day1/example.txt')

      const answer = measureDepthWindowIncreases(input)

      assert.equal(answer, 5)
    })

    it("input should return answer", () => {
      const input = readPuzzleInput('./src/day1/input.txt')

      const answer = measureDepthWindowIncreases(input)

      assert.equal(answer, 1822)
    })
  })
})
